Starship branding on 4.0 page

bretthayes opened the pull request:
Updates the Sourcegraph 4.0 page with Starship branding; updated hero ~and cta~

### Changelog
- Closes #5702 by updating the hero with starship branding and a few rearrangements
- Updated starship background gradients
- Added our new starship background svg for use with our `Background` component
- Adds a different variant for the starship logo which has less blur on the reflection
- Fixes a bug where the footer did not have a black background for the `navbar-dark` class
- Modified `ContentSection` and `CtaSection` components to inherit our Background component backgrounds
- Changed the API `button` prop on `CtaSection` to `ctaStyle` and updated all references
- Changed the API `color` prop on `ContentSection` to `background` and updated all references
- Changed the API `backButton` prop on `Hero` from `ReactNode` to an object with `link` and `text` to scope and standardize the component composition
- ~Added `CtaSection` to the 4.0 page with the starship branding~
- Cleaned up some components with more code reuse and removed our old `TrySourcegraph` component in lieu of our new standardized `CtaSection`
- Updated and replaced our dark nebulous gradient to use a code gradient and removed the image gradient background from our Background component

### Test

1. Ensure prettier has standardized the proposed changes.
2. Ensure page is neatly responsive

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202885951329736

st0nebraker reviewed (commented):
LGTM!
- What's a reflected image asset? I don't see the differences in this asset compared to #5703 
- I'm unclear what [Figma copy design](https://www.figma.com/file/lXvp8a2qoAU5L5sm4Woisu/STARSHIP-brand-assets?node-id=48%3A4907) is being used. Is the CTA section new on there? There's a bg image and other missing CTAs. I know this ticket is for the hero, but maybe the CTA can be added if it's a small lift?

bretthayes commented:
@st0nebraker I just named the asset with "reflected" since the dropshadow reflection on the "starship" text is reflected further rather than blurred like the asset in #5703. Open to renaming it to something else if you have an idea 💡 

For the CTA section at the bottom, I ended up adding it even though it wasn't included in the ticket. I added the new starship background to our Background component, and also modified our `ContentSection` and `CtaSection` component to inherit backgrounds. Yay for code reuse! 😄 So now we can tackle some more tech debt and replace some items around the codebase to use `ContentSection` with backgrounds instead of custom divs where we use custom backgrounds or the Background component. I also changed the API of `ContentSection` to use `background` instead of `color` as the prop name.

bretthayes commented:
@st0nebraker

Update:
- the `CtaSection` is temporarily removed for now on the 4.0 page
- I added a few improvements in [9159f25](https://github.com/sourcegraph/about/pull/5704/commits/9159f25e18fb100b046d2a7ffc5476dd6522a87b) with more code reuse. We're now using `Hero` and `CtaSection` more heavily now and I removed the old `TrySourcegraph` component in lieu of our new standardized `CtaSection` component from the DLS
- I changed the API of the `backButton` prop in our `Hero` component to make it more strict and standardized, so now you can pass an object with a `link` and `text` instead of always including the BackButton component

st0nebraker commented on src/components/Blog/PostLayout.tsx:
```diff
@@ -111,8 +101,20 @@ export const PostLayout: FunctionComponent<PostComponentProps> = ({
-        <ContentSection>
-            <TrySourcegraph stackedButtons={true} />
-        </ContentSection>
+        <CtaSection
+            centerContent={true}
+            title="Try Sourcegraph for free today"
+            description="You'll be searching your own code in 10 minutes. You can run it self-hosted (all of your code stays local and secure)."
+            cta1={{
+                text: 'Try Sourcegraph now',
+                link: '/get-started/self-hosted',
+                ctaStyle: 'primaryButton',
+            }}
+            cta2={{
+                text: 'Schedule a demo',
+                link: '/demo',
+                ctaStyle: 'outlineButton',
+            }}
+        />
```
Flagging that trading out the Blog CTA makes this section a lot longer than it was previously and on the DLS. I think on the blog it makes sense for this section to fit the blog's feel as it's different from the rest of the site. What're your thoughts?

<img width="955" alt="image" src="https://user-images.githubusercontent.com/59381432/187727521-3be4fc9d-e36d-4298-9a4b-c76ba20c5f30.png">

bretthayes replied:
Yeah I agree about fitting the blog's "feel." I think we can make this work with a few revisions. Working with Tracey atm to solidify it and will update shortly!

bretthayes replied:
@st0nebraker ok, all set now! Check out a blog post for the new slim width CtaSection variant. We also changed the blog content width to align with the DLS so we're all matched up now.

st0nebraker reviewed (commented):
Since this became a bit of a big refactor & the changes touch a few dozen pages I didn't fully test all the pages with our time crunch. I trust they were tested in development too.

Just a note below on a blog CTA UI discrepancy I noticed.

The added button tracking is *chefs kiss*

st0nebraker commented on src/components/CtaSection.tsx:
```diff
@@ -87,32 +105,39 @@ const Cta: FunctionComponent<Cta> = ({ text, icon, button = false, link, onClick
         <div
             className={classNames({
                 'tw-max-w-xl tw-mx-auto tw-text-center': centerContent,
                 'tw-grid tw-grid-cols-5': !centerContent,
```
```suggestion
                'tw-grid tw-grid-cols-5': !centerContent && !slimWidth,
```

bretthayes replied:
Oh, thanks for catching this! I just realized how this was spanning since I had the `CtaSection` in the PostLayout. So the full width of the background was contained within the content container. I moved this outside of it to the blog slug so it spans across the full width of the page properly now.

st0nebraker replied:
At the medium screen width I think it should start left aligning because it looks a bit off, no ?

<img width="816" alt="image" src="https://user-images.githubusercontent.com/59381432/187772288-33bb0700-7d3b-48d9-a8c0-cb46c11fe575.png">

bretthayes replied:
Yeah, I agree! With the grid approach, everything comes together at the centre so I added a modification to approach it with a combo of your max width suggestion and the grid with some padding on the description above tablet to get it as close as possible.

st0nebraker replied:
It's beautiful! 😻  Thanks!

st0nebraker commented on src/components/CtaSection.tsx:
```diff
@@ -87,32 +105,39 @@ const Cta: FunctionComponent<Cta> = ({ text, icon, button = false, link, onClick
             className={classNames({
                 'tw-max-w-xl tw-mx-auto tw-text-center': centerContent,
                 'tw-grid tw-grid-cols-5': !centerContent,
+                'tw-grid tw-grid-cols-12 tw-gap-sm tw-grid-flow-row tw-items-center': slimWidth,
```
The above suggestion & this should help the text align on the blog page

```suggestion
                'tw-max-w-[846px] tw-mx-auto tw-grid tw-grid-cols-2 tw-items-start': slimWidth,
```
BEFORE:
![image](https://user-images.githubusercontent.com/59381432/187766401-f0e2a74e-9c15-4216-9922-17f70823d0a6.png)
AFTER:
![image](https://user-images.githubusercontent.com/59381432/187766063-a6d6647f-b71a-429b-800d-986f2f2e36b0.png)

bretthayes replied:
This doesn't match the DLS spec but it's a similar approach using max width vs grid columns and grid gaps. We want the left content to stay within 4-5 columns. The buttons also need to be vertically centred with the left column content instead of using items-start.

bretthayes commented:
@st0nebraker I updated and replaced our dark nebulous gradient to use a code gradient and removed the image gradient background from our Background component so we don't lose any loading performance points here. See commit [2d25a10](https://github.com/sourcegraph/about/pull/5704/commits/2d25a10430793ef9d4f468749485d4d1b8a39f56). I'll implement the rest in another PR.

st0nebraker reviewed (approved):
Looks good! Woot woot for performance boosts nixing 🙅🏻  the images
